Realm/ObjectStore/object_store.cpp

```diff
@@ -346,35 +348,51 @@ bool ObjectStore::create_tables(Group *group, ObjectStore::Schema &target_schema
 bool ObjectStore::is_schema_at_version(Group *group, uint64_t version) {
     uint64_t old_version = get_schema_version(group);
     if (old_version > version && old_version != NotVersioned) {
-        throw ObjectStoreException(ObjectStoreException::Kind::RealmVersionGreaterThanSchemaVersion,
-                                   {{"old_version", to_string(old_version)}, {"new_version", to_string(version)}});
+        throw InvalidSchemaVersionException(old_version, version);
     }
-    return old_version != version;
+    return old_version == version;
 }
 
+bool ObjectStore::needs_update(Schema const& old_schema, Schema& schema) {
```

Comment:
this looks like it doesn't handle the set of properties changing, which will be needed if we want a `delete_if_migration_required` option